h2o-core/src/test/java/water/TestUtil.java

```diff
@@ -683,10 +688,20 @@ public static void assertVecEquals(Vec expecteds, Vec actuals, double delta) {
   }
 
   public static void assertVecEquals(String messagePrefix, Vec expecteds, Vec actuals, double delta) {
+    assertVecEquals(messagePrefix, expecteds, actuals, delta, false);
+  }
+
+  public static void assertVecEquals(String messagePrefix, Vec expecteds, Vec actuals, double delta, boolean relativeDelta) {
     assertEquals(expecteds.length(), actuals.length());
     for(int i = 0; i < expecteds.length(); i++) {
       final String message = messagePrefix + i + ": " + expecteds.at(i) + " != " + actuals.at(i) + ", chunkIds = " + expecteds.elem2ChunkIdx(i) + ", " + actuals.elem2ChunkIdx(i) + ", row in chunks = " + (i - expecteds.chunkForRow(i).start()) + ", " + (i - actuals.chunkForRow(i).start());
-      assertEquals(message, expecteds.at(i), actuals.at(i), delta);
+      double expectedVal = expecteds.at(i);
+      double actualVal = actuals.at(i);
+      double assertionDelta = delta;
+      if (relativeDelta) {
+        assertionDelta = Math.min(expectedVal, actualVal) * delta;
```

Comment:
How does this work for negative numbers or if the minimum is 0? Then assertionDelta = 0. Which means exact match.

Reply:
good point, updated, the whole thing with min() was wrong, we should check the delta relative to the expected value only and use absolute value of the expected